getnacelle/nacelle-lambda-tools
# Add locale normalization and dilithiumV2

### WHY are these changes introduced?

For both [[NC-1801](https://nacelle.atlassian.net/browse/NC-1801)] and [[NC-1802](https://nacelle.atlassian.net/browse/NC-1802)]
NB: Changes to this repo are not dependent on anything, but the changes in (this PR)[https://github.com/getnacelle/nacelle-admin-graphql/pull/32] are dependent on this PR.

### WHAT is this pull request doing?

This is adding the ability to index with [nacelle-admin-graphql](https://github.com/getnacelle/nacelle-admin-graphql) with dilithiumV2 as well as the `normalizeLocales` field. Both changes are behind a feature flag. The `isFeatureAllowed` function was created in [nacelle-lambda-tools](https://github.com/getnacelle/nacelle-lambda-tools) to determine whether the feature is allowed or not. Going forward the `isFeatureAllowed` function in nacelle-lambda-tools should be used for all feature flag situations.

### How to Test

Run nacelle-admin-graphql locally (with the changes in (this PR)[https://github.com/getnacelle/nacelle-admin-graphql/pull/32]), link this nacelle-lambda-tools repo and attempt to index with spaces that have/do not have the featureFlags "nacelleLocales" or "dilithiumV2". Having "nacelleLocales" should add a `normalizeLocales: true` field to the body of the request and having "dilithiumV2" should index via dilithium v2. 

You can also set the fields `FEATURE_DILITHIUM_V2: 'true'` and `FEATURE_NACELLE_LOCALES: 'true'` in the serverless.yml to cause a feature flag to be active.
